tests/src/test/scala/cats/tests/KleisliSuite.scala

```diff
@@ -39,13 +39,26 @@ class KleisliSuite extends CatsSuite {
   checkAll("Semigroupal[Kleisli[Option, Int, ?]]", SerializableTests.serializable(Semigroupal[Kleisli[Option, Int, ?]]))
 
   checkAll("Kleisli[(CSemi, ?), Int, ?]", CommutativeFlatMapTests[Kleisli[(CSemi, ?), Int, ?]].commutativeFlatMap[Int, Int, Int])
-  checkAll("CommutativeFlatMap[Kleisli[(CSemi, ?), Int, ?]]",SerializableTests.serializable(CommutativeFlatMap[Kleisli[(CSemi, ?), Int, ?]]))
+  checkAll("CommutativeFlatMap[Kleisli[(CSemi, ?), Int, ?]]", SerializableTests.serializable(CommutativeFlatMap[Kleisli[(CSemi, ?), Int, ?]]))
 
   checkAll("Kleisli[Option, Int, ?]", CommutativeMonadTests[Kleisli[Option, Int, ?]].commutativeMonad[Int, Int, Int])
-  checkAll("CommutativeMonad[Kleisli[Option, Int, ?]]",SerializableTests.serializable(CommutativeMonad[Kleisli[Option, Int, ?]]))
+  checkAll("CommutativeMonad[Kleisli[Option, Int, ?]]", SerializableTests.serializable(CommutativeMonad[Kleisli[Option, Int, ?]]))
 
   checkAll("Kleisli[Id, Int, ?]", CommutativeMonadTests[Kleisli[Id, Int, ?]].commutativeMonad[Int, Int, Int])
-  checkAll("CommutativeMonad[Kleisli[Id, Int, ?]]",SerializableTests.serializable(CommutativeMonad[Kleisli[Id, Int, ?]]))
+  checkAll("CommutativeMonad[Kleisli[Id, Int, ?]]", SerializableTests.serializable(CommutativeMonad[Kleisli[Id, Int, ?]]))
+
+  //TODO: Why can't type inference figure this out?
+  implicit val x: ErrorControl[Kleisli[Option, Int, ?], Kleisli[Id, Int, ?], Unit] =
+    Kleisli.catsErrorControlForKleisli[Option, Id, Int, Unit]
+
+  checkAll("Kleisli[Option, Int, ?]", ErrorControlTests[Kleisli[Option, Int, ?], Kleisli[Id, Int, ?], Unit].errorControl[Int])
```

Comment:
For some reason Scala can't seem to find an implicit `ErrorControl[Kleisli[Option, Int, ?], Kleisli[Id, Int, ?], Unit]` even though it is defined in the companion object of `Kleisli`. Anyone know a way to help scalac here?

Reply:
what if you call `Kleisli.catsErrorControlForKleisli[Option, Id, Int, Unit]` directly? does it complain that couldn't find `ErrorControl[Option, Id, Unit]`?

Reply:
Yeah it won't compile without it :/

Reply:
hmm, since StateT has exactly the same shape but passes with an implicit Monad[G], I wonder what if you add `Monad[G]` to `Kleisli` might help? 
Generally, when I get into this type of error (the implicit method is available and legit but scalac refuses to find it) I'll just throw in a `Lazy` and see if it helps. but that's not an option here. And I suspect that it's not the same scalac bug.

Reply:
Nope, I needed to add the extra implicit to WriterT and StateT as well, so it's the same problem.

Reply:
(╯°□°)╯︵ ┻━┻  time! 
Just to experiment, what if add them to the companion of ErrorControl?